### ISSUE TITLE
Not Working on some Android(10) devices 

### ISSUE THREAD

**Issue by mwaked** [AnchorFreePartner/hydrasdk-demo-android#71]
It's working fine on all devices but only on some android 10 it's totally not working and give me this error: 
Error: **g.b.i.m.n: com.anchorfree.vpnsdk.vpnservice.credentials.f: Captive Portal**

```
open fun handleError(e: Throwable
![Screenshot_20200717-150613](https://user-images.githubusercontent.com/33099498/87822421-1b6e4c00-c87a-11ea-81df-f192a5b2e230.png)
?) {
       
     when (e) {
            is NetworkRelatedException -> {
                showMessage(getString(R.string.text_check_intenet_connection))
            }
            is VpnException -> {
                when (e) {
                    is VpnPermissionRevokedException -> {
                        showMessage(getString(R.string.revoked_permission))
                    }
                    is VpnPermissionDeniedException -> {
                        showMessage(getString(R.string.cancel_permission))
                    }
                    is HydraVpnTransportException -> {
                        when (e.code) {
                            HydraVpnTransportException.HYDRA_ERROR_BROKEN -> {
                                showMessage(getString(R.string.connection_vpn_lost))
                            }
                            HydraVpnTransportException.HYDRA_DCN_BLOCKED_BW -> {
                                showMessage(getString(R.string.trafic_exceeded))
                            }
                            else -> {
                                showMessage(getString(R.string.error_vpn_transport))
                            }
                        }
                    }
                    else -> {
                        Log.d(TAG, "Error: $e")
                       // This is the error which happens on Android 10: g.b.i.m.n: com.anchorfree.vpnsdk.vpnservice.credentials.f: Captive Portal
                        showMessage("Error in VPN Service")
                    }
                }
            }
            is PartnerApiException -> {
                when (e.content) {
                    PartnerApiException.CODE_NOT_AUTHORIZED -> showMessage("User unauthorized")
                    PartnerApiException.CODE_TRAFFIC_EXCEED -> showMessage("Server unavailable")
                    else -> showMessage("Other error. Check PartnerApiException constants")
                }
            }
        }
    }
```

**Device Details:**

- Version: Android 10

- Device Name: Pixel 2

**Log:**
`2020-07-17 22:22:48.702 25470-25470/com.rket.reocketvpn D/USDK-CarrierVPN: ---------------------------------------------------------
2020-07-17 22:22:48.703 25470-25470/com.rket.reocketvpn D/USDK-CarrierVPN: | StartVPN: session: SessionConfig{virtualLocation='', config=null, dnsConfig=[com.anchorfree.sdk.j6.a$d@b888f22], appPolicy=AppPo |
2020-07-17 22:22:48.703 25470-25470/com.rket.reocketvpn D/USDK-CarrierVPN: | licy{policy=0, appList=[]}, extras={}, transport='hydra', reason='m_ui', sessionId='', vpnParams='VpnParams{dns1='8.8.8.8', dns2 |
2020-07-17 22:22:48.703 25470-25470/com.rket.reocketvpn D/USDK-CarrierVPN: | ='8.8.4.4', routes=[Route{route='128.0.0.0', mask=1}, Route{route='0.0.0.0', mask=1}]}', privateGroup='', keepOnReconnect='false |
2020-07-17 22:22:48.703 25470-25470/com.rket.reocketvpn D/USDK-CarrierVPN: | ', captivePortalBlockBypass='false'}                                                                                             |
2020-07-17 22:22:48.703 25470-25470/com.rket.reocketvpn D/USDK-CarrierVPN: ---------------------------------------------------------
2020-07-17 22:22:48.756 25470-25542/com.rket.reocketvpn D/USDK-RemoteVpn: Start vpn and check bound                                                                                                       
2020-07-17 22:22:48.758 25470-26310/com.rket.reocketvpn D/USDK-RemoteVpn: Call remote service to start                                                                                                    
2020-07-17 22:22:48.840 25470-25803/com.rket.reocketvpn D/USDK-RemoteVpn: Change state to CONNECTING_PERMISSIONS                                                                                          
2020-07-17 22:22:48.840 25470-25797/com.rket.reocketvpn D/USDK-UnifiedSDK: Vpn state changed to VPNState{state='CONNECTING_PERMISSIONS}                                                                    
2020-07-17 22:22:48.925 25470-25522/com.rket.reocketvpn D/FA: Connected to remote service
2020-07-17 22:22:49.631 25470-25470/com.rket.reocketvpn E/FirebaseInstanceId: binding to the service failed
2020-07-17 22:22:50.604 25470-25501/com.rket.reocketvpn D/USDK-RemoteVpn: Change state to CONNECTING_CREDENTIALS                                                                                          
2020-07-17 22:22:50.605 25470-25797/com.rket.reocketvpn D/USDK-UnifiedSDK: Vpn state changed to VPNState{state='CONNECTING_CREDENTIALS}                                                                    
2020-07-17 22:22:50.707 25470-25470/com.rket.reocketvpn D/BaseActivity: Error: g.b.i.m.n: com.anchorfree.vpnsdk.vpnservice.credentials.f: Captive Portal
2020-07-17 22:22:50.717 25470-25522/com.rket.reocketvpn W/FA: Value is too long; discarded. Value kind, name, value length: param, error, 157
2020-07-17 22:22:51.704 25470-25501/com.rket.reocketvpn D/USDK-CredentialsConten: loadCredsTokenSource equal new. skip set                                                                                        
2020-07-17 22:22:51.748 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: ---------------------------------------------------------
2020-07-17 22:22:51.748 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | Trackaction: connection_starttimestamp:1595013771747props: Bundle[{device_manufacturer=Google, has_telephone=true, duration=2    |
2020-07-17 22:22:51.748 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | 937, partner_carrier=301301_rocketvpn, catime=1595013768764, is_ipv6_only=0, locale=US, af_hash=MzAxMzAxX3JvY2tldHZwbl9kZDE4OTE1 |
2020-07-17 22:22:51.748 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | ZmVkNGE1MmFi, protocol=hydra, memory_remains=1762000896, reason=m_ui, app_version=1.5.6, device_language=en, sdk_version=3.3.1,  |
2020-07-17 22:22:51.748 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | app_release=114, uid=10329, caid=66AC925654F64FE690AD468F07EF63E1, time_zone=-0400, error=Captive Portal, model=Pixel 2, notes={ |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | "error_details":"com.anchorfree.vpnsdk.vpnservice.credentials.f: Captive Portal\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptive |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | PortalChecker.c(SDKCaptivePortalChecker.java:62)\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.b(SDKCaptivePorta |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | lChecker.java:22)\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker$a.a(SDKCaptivePortalChecker.java:80)\n\tat com.a |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | nchorfree.vpnsdk.vpnservice.credentials.DefaultCaptivePortalChecker.h(DefaultCaptivePortalChecker.java:136)\n\tat com.anchorfree |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | .vpnsdk.vpnservice.credentials.DefaultCaptivePortalChecker.a(DefaultCaptivePortalChecker.java:76)\n\tat com.anchorfree.vpnsdk.re |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | connect.SDKCaptivePortalChecker.d(SDKCaptivePortalChecker.java:72)\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | .e(SDKCaptivePortalChecker.java:43)\n\tat com.anchorfree.vpnsdk.reconnect.h.a(Unknown Source:13)\n\tat g.b.d.j$e.run(Task.java:9 |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | 14)\n\tat g.b.d.c$b.execute(BoltsExecutors.java:108)\n\tat g.b.d.j.i(Task.java:905)\n\tat g.b.d.j.a(Task.java:34)\n\tat g.b.d.j$ |
2020-07-17 22:22:51.749 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | a.b(Task.java:681)\n\tat g.b.d.j$a.a(Task.java:677)\n\tat g.b.d.j.G(Task.java:999)\n\tat g.b.d.j.J(Task.java:1037)\n\tat g.b.d.k |
2020-07-17 22:22:51.750 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | .g(TaskCompletionSource.java:43)\n\tat g.b.d.k.d(TaskCompletionSource.java:66)\n\tat g.b.d.j$i.run(Task.java:369)\n\tat java.uti |
2020-07-17 22:22:51.750 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | l.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run |
2020-07-17 22:22:51.750 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | (ThreadPoolExecutor.java:641)\n\tat java.lang.Thread.run(Thread.java:919)\n"}, carrier=Ooredoo, sd_in_tunnel=0, connection_type= |
2020-07-17 22:22:51.750 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | WiFi, af_platform=android, android_version_name=10, memory_total=3834605568, app_name=com.rket.reocketvpn, hydra_version=release |
2020-07-17 22:22:51.750 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | -hydra-v5.0.4-30-g67df40c, sdk_version_code=403503, android_sdk_int=29, error_code=4, app_build=-754555370}]                     |
2020-07-17 22:22:51.750 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: ---------------------------------------------------------
2020-07-17 22:22:51.762 25470-25663/com.rket.reocketvpn D/USDK-UCRService: registerContentObserver onChange                                                                                                
2020-07-17 22:22:51.762 25470-25663/com.rket.reocketvpn D/USDK-UCRService: queueUpload                                                                                                                     
2020-07-17 22:22:55.486 25470-25501/com.rket.reocketvpn D/USDK-RemoteVpn: Change state to PAUSED                                                                                                          
2020-07-17 22:22:55.486 25470-25797/com.rket.reocketvpn D/USDK-UnifiedSDK: Vpn state changed to VPNState{state='PAUSED}                                                                                    
2020-07-17 22:22:55.500 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: ---------------------------------------------------------
2020-07-17 22:22:55.500 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | Trackaction: connection_start_detailedtimestamp:1595013775500props: Bundle[{device_manufacturer=Google, has_telephone=true, d    |
2020-07-17 22:22:55.500 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | uration=2937, partner_carrier=301301_rocketvpn, catime=1595013768764, is_ipv6_only=0, locale=US, af_hash=MzAxMzAxX3JvY2tldHZwbl9 |
2020-07-17 22:22:55.500 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | kZDE4OTE1ZmVkNGE1MmFi, protocol=hydra, memory_remains=1758511104, reason=m_ui, app_version=1.5.6, device_language=en, sdk_versio |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | n=3.3.1, app_release=114, uid=10329, caid=66AC925654F64FE690AD468F07EF63E1, time_zone=-0400, error=Captive Portal, model=Pixel 2 |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | , notes={"network_availability_test":[{"type":"network interface","result":"wifi"},{"type":"http certificate","url":"https:\/\/l |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | inkedin.com","result":"ok"},{"type":"captive portal","url":"https:\/\/google.com\/generate_204","result":"ok"}],"error_details": |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | "com.anchorfree.vpnsdk.vpnservice.credentials.f: Captive Portal\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.c( |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | SDKCaptivePortalChecker.java:62)\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.b(SDKCaptivePortalChecker.java:22 |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | )\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker$a.a(SDKCaptivePortalChecker.java:80)\n\tat com.anchorfree.vpnsdk |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | .vpnservice.credentials.DefaultCaptivePortalChecker.h(DefaultCaptivePortalChecker.java:136)\n\tat com.anchorfree.vpnsdk.vpnservi |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | ce.credentials.DefaultCaptivePortalChecker.a(DefaultCaptivePortalChecker.java:76)\n\tat com.anchorfree.vpnsdk.reconnect.SDKCapti |
2020-07-17 22:22:55.501 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | vePortalChecker.d(SDKCaptivePortalChecker.java:72)\n\tat com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.e(SDKCaptivePor |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | talChecker.java:43)\n\tat com.anchorfree.vpnsdk.reconnect.h.a(Unknown Source:13)\n\tat g.b.d.j$e.run(Task.java:914)\n\tat g.b.d. |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | c$b.execute(BoltsExecutors.java:108)\n\tat g.b.d.j.i(Task.java:905)\n\tat g.b.d.j.a(Task.java:34)\n\tat g.b.d.j$a.b(Task.java:68 |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | 1)\n\tat g.b.d.j$a.a(Task.java:677)\n\tat g.b.d.j.G(Task.java:999)\n\tat g.b.d.j.J(Task.java:1037)\n\tat g.b.d.k.g(TaskCompletio |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | nSource.java:43)\n\tat g.b.d.k.d(TaskCompletionSource.java:66)\n\tat g.b.d.j$i.run(Task.java:369)\n\tat java.util.concurrent.Thr |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | eadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecu |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | tor.java:641)\n\tat java.lang.Thread.run(Thread.java:919)\n"}, carrier=Ooredoo, sd_in_tunnel=0, connection_type=WiFi, af_platfor |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | m=android, android_version_name=10, memory_total=3834605568, app_name=com.rket.reocketvpn, hydra_version=release-hydra-v5.0.4-30 |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | -g67df40c, sdk_version_code=403503, details=[], android_sdk_int=29, error_code=4, app_build=-754555370, network_availability=1.0 |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: | }]                                                                                                                               |
2020-07-17 22:22:55.502 25470-25691/com.rket.reocketvpn D/USDK-EventHelper: ---------------------------------------------------------
2020-07-17 22:22:55.514 25470-25663/com.rket.reocketvpn D/USDK-UCRService: registerContentObserver onChange                                                                                                
2020-07-17 22:22:55.514 25470-25663/com.rket.reocketvpn D/USDK-UCRService: queueUpload                                                                                                                     
2020-07-17 22:23:05.520 25470-25710/com.rket.reocketvpn D/USDK-UCRService: performUpload                                                                                                                   
2020-07-17 22:23:05.529 25470-25710/com.rket.reocketvpn D/USDK-UCRService: Transport upload: sdk                                                                                                           
2020-07-17 22:23:05.529 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: upload                                                                                                                          
2020-07-17 22:23:05.538 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: ---------------------------------------------------------
2020-07-17 22:23:05.539 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | Perform Request data: {"a":"1","event":"connection_start","properties":{"af_platform":"android","reason":"m_ui","notes":"{\"erro |
2020-07-17 22:23:05.539 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | r_details\":\"com.anchorfree.vpnsdk.vpnservice.credentials.f: Captive Portal\\at com.anchorfree.vpnsdk.reconnect.SDKCaptivePorta |
2020-07-17 22:23:05.539 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | lChecker.c(SDKCaptivePortalChecker.java:62)\\at com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.b(SDKCaptivePortalChecke |
2020-07-17 22:23:05.539 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | r.java:22)\\at com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker$a.a(SDKCaptivePortalChecker.java:80)\\at com.anchorfree.v |
2020-07-17 22:23:05.540 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | pnsdk.vpnservice.credentials.DefaultCaptivePortalChecker.h(DefaultCaptivePortalChecker.java:136)\\at com.anchorfree.vpnsdk.vpnse |
2020-07-17 22:23:05.540 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | rvice.credentials.DefaultCaptivePortalChecker.a(DefaultCaptivePortalChecker.java:76)\\at com.anchorfree.vpnsdk.reconnect.SDKCapt |
2020-07-17 22:23:05.540 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | ivePortalChecker.d(SDKCaptivePortalChecker.java:72)\\at com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.e(SDKCaptivePort |
2020-07-17 22:23:05.540 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | alChecker.java:43)\\at com.anchorfree.vpnsdk.reconnect.h.a(Unknown Source:13)\\at g.b.d.j$e.run(Task.java:914)\\at g.b.d.c$b.exe |
2020-07-17 22:23:05.540 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | cute(BoltsExecutors.java:108)\\at g.b.d.j.i(Task.java:905)\\at g.b.d.j.a(Task.java:34)\\at g.b.d.j$a.b(Task.java:681)\\at g.b.d. |
2020-07-17 22:23:05.541 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | j$a.a(Task.java:677)\\at g.b.d.j.G(Task.java:999)\\at g.b.d.j.J(Task.java:1037)\\at g.b.d.k.g(TaskCompletionSource.java:43)\\at  |
2020-07-17 22:23:05.541 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | g.b.d.k.d(TaskCompletionSource.java:66)\\at g.b.d.j$i.run(Task.java:369)\\at java.util.concurrent.ThreadPoolExecutor.runWorker(T |
2020-07-17 22:23:05.541 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | hreadPoolExecutor.java:1167)\\at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)\\at java.lang.T |
2020-07-17 22:23:05.541 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | hread.run(Thread.java:919)\\"}","app_version":"1.5.6","is_ipv6_only":"0","locale":"US","error":"Captive Portal","device_manufact |
2020-07-17 22:23:05.541 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | urer":"Google","duration":"2937","uid":10329,"protocol":"hydra","hydra_version":"release-hydra-v5.0.4-30-g67df40c","catime":"159 |
2020-07-17 22:23:05.542 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | 5013768764","caid":"66AC925654F64FE690AD468F07EF63E1","sdk_version":"3.3.1","model":"Pixel 2","memory_total":3834605568,"af_hash |
2020-07-17 22:23:05.542 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | ":"MzAxMzAxX3JvY2tldHZwbl9kZDE4OTE1ZmVkNGE1MmFi","memory_remains":1762000896,"seq_no":0,"app_release":114,"connection_type":"WiF |
2020-07-17 22:23:05.542 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | i","android_version_name":"10","has_telephone":true,"sd_in_tunnel":"0","time_zone":"-0400","app_name":"com.rket.reocketvpn","car |
2020-07-17 22:23:05.542 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | rier":"Ooredoo","sdk_version_code":"403503","partner_carrier":"301301_rocketvpn","app_build":-754555370,"android_sdk_int":29,"er |
2020-07-17 22:23:05.542 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | ror_code":"4","time":1595013771747,"device_language":"en"}}{"a":"2","event":"connection_start_detailed","properties":{"af_platf  |
2020-07-17 22:23:05.542 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | orm":"android","reason":"m_ui","notes":"{\"network_availability_test\":[{\"type\":\"network interface\",\"result\":\"wifi\"},{\" |
2020-07-17 22:23:05.542 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | type\":\"http certificate\",\"url\":\"https:\\/\\/linkedin.com\",\"result\":\"ok\"},{\"type\":\"captive portal\",\"url\":\"https |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | :\\/\\/google.com\\/generate_204\",\"result\":\"ok\"}],\"error_details\":\"com.anchorfree.vpnsdk.vpnservice.credentials.f: Capti |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | ve Portal\\at com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.c(SDKCaptivePortalChecker.java:62)\\at com.anchorfree.vpns |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | dk.reconnect.SDKCaptivePortalChecker.b(SDKCaptivePortalChecker.java:22)\\at com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChec |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | ker$a.a(SDKCaptivePortalChecker.java:80)\\at com.anchorfree.vpnsdk.vpnservice.credentials.DefaultCaptivePortalChecker.h(DefaultC |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | aptivePortalChecker.java:136)\\at com.anchorfree.vpnsdk.vpnservice.credentials.DefaultCaptivePortalChecker.a(DefaultCaptivePorta |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | lChecker.java:76)\\at com.anchorfree.vpnsdk.reconnect.SDKCaptivePortalChecker.d(SDKCaptivePortalChecker.java:72)\\at com.anchorf |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | ree.vpnsdk.reconnect.SDKCaptivePortalChecker.e(SDKCaptivePortalChecker.java:43)\\at com.anchorfree.vpnsdk.reconnect.h.a(Unknown  |
2020-07-17 22:23:05.543 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | Source:13)\\at g.b.d.j$e.run(Task.java:914)\\at g.b.d.c$b.execute(BoltsExecutors.java:108)\\at g.b.d.j.i(Task.java:905)\\at g.b. |
2020-07-17 22:23:05.544 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | d.j.a(Task.java:34)\\at g.b.d.j$a.b(Task.java:681)\\at g.b.d.j$a.a(Task.java:677)\\at g.b.d.j.G(Task.java:999)\\at g.b.d.j.J(Tas |
2020-07-17 22:23:05.544 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | k.java:1037)\\at g.b.d.k.g(TaskCompletionSource.java:43)\\at g.b.d.k.d(TaskCompletionSource.java:66)\\at g.b.d.j$i.run(Task.java |
2020-07-17 22:23:05.544 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | :369)\\at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)\\at java.util.concurrent.ThreadPoolExe |
2020-07-17 22:23:05.544 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | cutor$Worker.run(ThreadPoolExecutor.java:641)\\at java.lang.Thread.run(Thread.java:919)\\"}","app_version":"1.5.6","is_ipv6_only |
2020-07-17 22:23:05.544 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | ":"0","locale":"US","error":"Captive Portal","device_manufacturer":"Google","duration":"2937","uid":10329,"protocol":"hydra","hy |
2020-07-17 22:23:05.545 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | dra_version":"release-hydra-v5.0.4-30-g67df40c","catime":"1595013768764","caid":"66AC925654F64FE690AD468F07EF63E1","sdk_version" |
2020-07-17 22:23:05.545 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | :"3.3.1","model":"Pixel 2","details":"[]","memory_total":3834605568,"af_hash":"MzAxMzAxX3JvY2tldHZwbl9kZDE4OTE1ZmVkNGE1MmFi","me |
2020-07-17 22:23:05.545 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | mory_remains":1758511104,"seq_no":1,"app_release":114,"connection_type":"WiFi","network_availability":"1.0","android_version_nam |
2020-07-17 22:23:05.545 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | e":"10","has_telephone":true,"sd_in_tunnel":"0","time_zone":"-0400","app_name":"com.rket.reocketvpn","carrier":"Ooredoo","sdk_ve |
2020-07-17 22:23:05.546 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | rsion_code":"403503","partner_carrier":"301301_rocketvpn","app_build":-754555370,"android_sdk_int":29,"error_code":"4","time":15 |
2020-07-17 22:23:05.546 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: | 95013775500,"device_language":"en"}}                                                                                             |
2020-07-17 22:23:05.546 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: ---------------------------------------------------------
2020-07-17 22:23:05.549 25470-25710/com.rket.reocketvpn D/USDK-TelemetryUrlProvi: Return null url due to wrong state: VPNState{state='PAUSED}                                                                     
2020-07-17 22:23:05.549 25470-25710/com.rket.reocketvpn D/USDK-DefaultTrackerTra: Provider returned empty url. Skip upload                                                                                        `

![Screenshot_20200717-150613](https://user-images.githubusercontent.com/33099498/87822498-3e006500-c87a-11ea-94e7-91643e8566b1.png)

**Comment by timoshenkoav** [comment]
Hello,
Do you have custom network security config in the app?

**Comment by mwaked** [comment]
Hello, 
Yes i have

```
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
    <base-config cleartextTrafficPermitted="true">
        <trust-anchors>
            <certificates src="system" />
        </trust-anchors>
    </base-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">whatismyip.akamai.com</domain>
    </domain-config>
</network-security-config>
```

**Comment by timoshenkoav** [comment]
try to add this also

`
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">connectivitycheck.gstatic.com</domain>
    </domain-config>
`

**Comment by mwaked** [comment]
I added it but still the same error happens

```
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
    <base-config cleartextTrafficPermitted="true">
        <trust-anchors>
            <certificates src="system" />
        </trust-anchors>

    </base-config>
    <domain-config cleartextTrafficPermitted="true">
        <domain includeSubdomains="true">connectivitycheck.gstatic.com</domain>
        <domain includeSubdomains="true">whatismyip.akamai.com</domain>
    </domain-config>
</network-security-config>
```

**Comment by timoshenkoav** [comment]
for some reason you device reports network has NetworkCapabilities.NET_CAPABILITY_CAPTIVE_PORTAL
could you please share all adb logcat output as a file?

**Comment by mwaked** [comment]
Sure

**Comment by mwaked** [comment]
Sure, check the file please
[log.txt](https://github.com/AnchorFreePartner/hydrasdk-demo-android/files/4939931/log.txt)

**Comment by timoshenkoav** [comment]
its not complete logs, please dont apply any filter to adb logcat output

**Comment by mwaked** [comment]
log file without any filter
[log.txt](https://github.com/AnchorFreePartner/hydrasdk-demo-android/files/4939963/log.txt)

**Comment by mwaked** [comment]
Please check
[log.txt](https://github.com/AnchorFreePartner/hydrasdk-demo-android/files/4939979/log.txt)

**Comment by timoshenkoav** [comment]
ok, as i can see in logs. you have 2 networks
wifi - without captive portal
cellular - with captive portal
i will investigate the issue. thanks for the report

**Comment by mwaked** [comment]
Great,
you are welcome and thank you too

**Comment by timoshenkoav** [comment]
Hello @mwaked ,
Please update sdk to just released version: 3.3.2
